### PR TITLE
feat: replace the "Edit this page" link by 3 icons

### PR DESCRIPTION
**Pull request by jeromecambon**
"Edit this page" button is replaced by 3 icons:
- Edit this page
- View page history
- View contribution guide

closes https://github.com/bonitasoft/bonita-documentation-theme/issues/145

### CODE REVIEW

**Comment by benjaminParisel** [comment]
A small comments (already here in actual code), but when are in small screen (mobile, or when you split your screen with another window), this button bar is not display. 

It could be nice if we can keep, at least the edition button), always visible.

**Comment by tbouffard** [comment]
### Light Theme
image




### Dark Theme
image

File: preview-src/admonitions.adoc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 = Admonitions
+:page-editable: true
```
Comment (tbouffard) [inline preview-src/admonitions.adoc:2]: Why did you change all pages to add the edit buttons? This is a lot of change and doesn't bring nothing
I suggest we revert it and we create 2 pages in the test sections to have one page with edit buttons and another one without buttons. We can keep the edit buttons on the index page.
What do you think about this proposal?

File: preview-src/ui-model.yml [hunk 2]
```diff
@@ -68,7 +68,7 @@ page:
   version: 'dev'
   displayVersion: 'dev'
   module: ROOT
-  editUrl: https://example.com/project-bonita/blob/master/index.adoc
+  editUrl: https://example.com/project-bonita/edit/master/index.adoc
```
Comment (tbouffard) [inline preview-src/ui-model.yml:71]: ℹ️ looks fine, the github url are something like https://github.com/bonitasoft/bonita-doc/edit/2022.2/modules/ROOT/pages/index.adoc

File: src/helpers/edit2var.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+'use strict'
+
+module.exports = (editUrl, type) => {
```
Comment (tbouffard) [inline src/helpers/edit2var.js:3]: I guess you take the implementation from fedora.
I suggest that
- we add a comment to provide the link of the original source (as a reminder or if we need to improve the function in the future
- we remove the type parameter which is useless here

File: src/partials/page-additional-icons.hbs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<a class="page-icon" href="{{{edit2var page.editUrl "history"}}}">
+  <img src="_/img/history.svg" title="History of this page" alt="History" height="17">
+</a>
+<a class="page-icon" href="https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc">
+  <img src="_/img/info.svg" title="How to contribute" alt="Contribution" height="17">
```
Comment (tbouffard) [inline src/partials/page-additional-icons.hbs:5]: ❓ about the icons, are they from an orginal design you did? If not can you put in comment where you find them?
⚠️ We must ensure that they are free to use or they are released under a license that allows us to integrate them.

**Review by tbouffard — CHANGES_REQUESTED** [review]
This PR globally looks good but they are some points that we should talk about

File: preview-src/admonitions.adoc (hunk 1 above)
Comment (jeromecambon) [reply preview-src/admonitions.adoc:2]: I did that on pupose, to mimic the actual site.
This allowed me to find a bug with Message Blocks (Out of support, ...). So IMHO it's better to keep this.
We could have a single page without the icons for testing purpose though.

**Comment by jeromecambon** [comment]
> A small comments (already here in actual code), but when are in small screen (mobile, or when you split your screen with another window), this button bar is not display.
> 
> It could be nice if we can keep, at least the edition button), always visible.

Yes I saw this also... I would suggest to create a separate issue for this (since it is not bring by this PR). I'm afraid it may be difficult to fix...

File: preview-src/admonitions.adoc (hunk 1 above)
Comment (tbouffard) [reply preview-src/admonitions.adoc]: You can add the edit buttons to the tests page that include the "custom message" block or "next versions" block instead. These 2 messages are intended to be used in page that always have the buttons.

File: src/partials/page-additional-icons.hbs (hunk 4 above)
Comment (jeromecambon) [reply src/partials/page-additional-icons.hbs:5]: I found them on the web, I checked thay didn't have a license and are free to use. Not sure it will be easy for me to find them again...

**Comment by tbouffard** [comment]
ℹ️ About the "edit this page", this is the behavior of the Antora Default UI: https://antora.gitlab.io/antora-ui-default/
I guess they suppose that people don't want to edit on mobile 😸 

image

File: preview-src/admonitions.adoc (hunk 1 above)
Comment (tbouffard) [reply preview-src/admonitions.adoc]: After discussions with Jerome, we decided to
- update the test pages involving block that are always marked as editable in production
- add a dedicated page for testing the display when the page is not editable
- make all other pages editable except the one for "out of support" which is never set as editable in production

Comment (tbouffard) [reply preview-src/admonitions.adoc:2]: done

File: src/helpers/edit2var.js (hunk 3 above)
Comment (tbouffard) [reply src/helpers/edit2var.js]: done in a41c8513

**Review by tbouffard — APPROVED** [review]
LGTM
